pkg/ottl/parser.go

```diff
@@ -27,6 +27,29 @@ type Parser[K any] struct {
 	telemetrySettings component.TelemetrySettings
 }
 
+// Statement is an interface that represents a parsed OTTL Statement
+type Statement[K any] interface {
+	// Execute is a function that will execute the statement's function if the statement's condition is met.
+	// Returns true if the function was run, returns false otherwise.
+	// If the statement contains no condition, the function will run and true will be returned.
+	// In addition, the functions return value is always returned.
+	Execute(ctx K) (any, bool)
+}
+
+type standardStatement[K any] struct {
+	function  ExprFunc[K]
+	condition boolExpressionEvaluator[K]
+}
+
+func (s standardStatement[K]) Execute(ctx K) (any, bool) {
+	condition := s.condition(ctx)
+	var result any
+	if condition {
+		result = s.function(ctx)
+	}
+	return result, condition
+}
```

Comment:
I like this, but I would do a bit different:

```suggestion
type Statement[K any] struct {
	function  ExprFunc[K]
	condition boolExpressionEvaluator[K]
}

// Execute is a function that will execute the statement's function if the statement's condition is met.
// Returns true if the function was run, returns false otherwise.
// If the statement contains no condition, the function will run and true will be returned.
// In addition, the functions return value is always returned.
func (s *Statement[K]) Execute(ctx K) (any, bool) {
	condition := s.condition(ctx)
	var result any
	if condition {
		result = s.function(ctx)
	}
	return result, condition
}
```

Or, add a private func to `Statement` to disallow others to implement. The motivation is that we don't expect others to implement this interface, and we may want to extend in the future.

Reply:
It's true, we don't expect others to implement it.  I will make this change.

Reply:
@bogdandrutu done